allennlp/models/ensemble/ensemble.py

```diff
@@ -0,0 +1,62 @@
+from typing import Dict, List
+
+import torch
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.params import Params
+from allennlp.models.model import Model
+from allennlp.models.model import remove_pretrained_embedding_params
+
+class Ensemble(Model):
+    """
+    An ensemble runs multiple instances of a model and selects an answer from the subresults via some
+    ensembling strategy.
+
+    Ensembles differ from most models in that they do not have a vocabulary or weights of their own
+    (instead they rely on the vocabulary and weights from submodels) and they are not directly
```

Comment:
There's no reason they shouldn't be directly trainable.  I don't think we need to include this.  You _could_ say something like "they are not typically trained jointly (but you can if you want to)."

Reply:
If you're instantiating a `Trainer` yourself, yes, you definitely could train this, as long as your ensemble included a `loss` key in the output.  And if you implement an `Ensemble` subclass yourself that can handle instantiating things `from_params` correctly, you could train this with `allennlp train` just fine.